app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/picture/PictureFragmentOnlineTest.kt

```diff
@@ -45,45 +48,61 @@ class PictureFragmentOnlineTest {
     fun setUp() {
         storage = Firebase.storage
         storage.useEmulator("10.0.2.2", 9199)
-        emptyFirebaseStorage(storage)
-
-        // Empty Local Files
         emptyLocalFiles(picturesDir)
 
-        // Launch the fragment
-        onView(ViewMatchers.withId(R.id.drawer_layout))
+        // Open the drawer
+        onView(withId(R.id.drawer_layout))
             .perform(DrawerActions.open())
-        onView(ViewMatchers.withId(R.id.pictureFragment))
+        onView(withId(R.id.roadBookFragment))
             .perform(ViewActions.click())
 
+        // Click on one of the roadbook
+        val destID = DestinationRecords.RECORDS[2].destID
+        onView(ViewMatchers.withText(destID)).perform(ViewActions.click())
+
+        // Go to the picture fragment
+        onView(withId(R.id.viewPager)).perform(swipeLeft())
+        onView(withId(R.id.viewPager)).perform(swipeLeft())
+        onView(withId(R.id.viewPager)).perform(swipeLeft())
+
         // Wait for the camera to open
         Thread.sleep(TIME_WAIT_SHUTTER)
     }
 
+    @After
+    fun tearDown() {
+        runBlocking { emptyFirebaseStorage(storage.reference) }
+        emptyLocalFiles(picturesDir)
+    }
+
 
     @Test
     fun testUploadFileCorrectly() {
         val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
 
-        val takePictureButton = device.findObject(UiSelector().description("Shutter"))
-        takePictureButton.click()
+        // Click on the shutter button
+        device.executeShellCommand("input keyevent 27")
```

Comment:
The keyevents constants are also shared between the offline and online tests. They could be defined in a Utils class.